e2e-tests/runner/examples/connectors/PhantomSolanaConnector/utils/waitForExtensions.ts

```diff
@@ -0,0 +1,46 @@
+import { setTimeout } from 'node:timers/promises';
+import type { BrowserContext } from '@playwright/test';
+
+export async function waitForExtensions(
+  context: BrowserContext,
+  extensions: Record<
+    string,
+    {
+      id: string;
+      version: string;
+    }
+  >,
+  maxAttempts = 5,
+  attempt = 0,
+): Promise<boolean> {
+  if (!context) throw new Error('BrowserContext is required.');
+  if (!extensions || typeof extensions !== 'object') {
+    throw new Error('Invalid extensions object provided.');
+  }
+
+  console.log(`Checking extensions (Attempt ${attempt + 1}/${maxAttempts})...`);
+  const pages = context.pages();
+
+  if (!pages.length) {
+    console.warn('No pages found in the context. Retrying...');
+  }
+
+  const phantomPage = pages.find((page) =>
+    page.url().includes(extensions.phantom?.id),
+  );
+
```

Comment:
I don't like empty lines inside methods/functions... but that's my personal opinion :D :D :D